Add throughput and latency documentation (#6910)
opensearch-project/documentation-website: _benchmark/user-guide/concepts/time-latency.md

---
layout: default
title: Throughput, latency, and service time
nav_order: 10
parent: Concepts
grand_parent: User Guide
---

# Throughput, latency, and service time

At the end of each test, OpenSearch Benchmark produces a table that summarizes the following: 

- [Took time](#took-time)
- [Service time](#service-time) 

> - Throughput

Reviewer: That would group the time-based metrics together

> - Throughput

Reviewer: Remove throughput on line 16

- [Latency](#latency)
- [Processing time]
- The error rate for each completed task or OpenSearch operation.

> The following diagram illustrates how each component of the table is measured during the life cycle of a request from an OpenSearch cluster to the OpenSearch client:

Reviewer: "request from an OpenSearch cluster to the OpenSearch client" --> "request involving the OpenSearch cluster, the OpenSearch client, and OpenSearch Benchmark"

> The following diagram illustrates how each component of the table is measured during the life cycle of a request from an OpenSearch cluster to the OpenSearch client:

Author: ```suggestion
The following diagram illustrates how each component of the table is measured during the life cycle of a request request involving the OpenSearch cluster, the OpenSearch client, and OpenSearch Benchmark:
```

OpenSearch Benchmark makes function calls to `opensearch-py` to communicate with an OpenSearch cluster. OpenSearch Benchmark tracks the amount of time between when the `opensearch-py` client sends a request and receives a response from the OpenSearch cluster and considers this to be the service time. Unlike the traditional definition of service time, the OpenSearch Benchmark definition of service time includes overhead, such as network latency, load balancer overhead, or deserialization/serialization. The following image highlights the differences between the traditional definition of service time and the OpenSearch Benchmark definition of service time.

<img src="{{site.url}}{{site.baseurl}}/images/benchmark/service-time.png" alt="">

## Latency

There are two types of testing modes when using OpenSearch Benchmark, which affects the definition of latency:

- [Benchmarking mode](#benchmarking-mode): Latency is measured the same as service time.

> - [Throughput-throttled mode](#throughput-throttled-mode): Latency is service-time plus the time a request spends waiting in the queue.

Reviewer: I like how concise this section is. But I think we can reorder it to make it easier to understand. 
1. First sentence cam be rephrased to "There are two types of testing modes that can be activated by the `target-throughput` parameter in OpenSearch Benchmark. Both testing modes affect the definition of latency. 
2. Let's put the Benchmarking Mode section before the Throughput-Throttled Mode. 
3. We can encapsulate Benchmarking Mode and Throughput-Throttled Mode under a larger header called "Throughput"
